langchain/src/callbacks/tracers.ts

```diff
@@ -98,43 +97,45 @@ export abstract class BaseTracer extends BaseCallbackHandler {
     }
   }
 
-  protected _startTrace(runId: string, run: LLMRun | ChainRun | ToolRun) {
+  protected _startTrace(run: LLMRun | ChainRun | ToolRun) {
     this.executionOrder += 1;
 
-    if (this.stack.length > 0) {
-      if (
-        !(
-          this.stack.at(-1)?.type === "tool" ||
-          this.stack.at(-1)?.type === "chain"
-        )
-      ) {
-        throw new Error("Nested run can only be logged for tool or chain");
+    if (run.parent_uuid) {
+      const parentRun = this.runMap.get(run.parent_uuid);
+      if (parentRun) {
+        if (!(parentRun.type === "tool" || parentRun.type === "chain")) {
+          throw new Error("Caller run can only be a tool or chain");
```

Comment:
I know this was already here, but what is the intended result of throwing this error? Not tracking this run?